doc/manual/src/protocols/signature.md

```diff
@@ -0,0 +1,49 @@
+# Signature
+
+In Nix, signatures are used to trust store paths using asymmetric cryptography,
+in this instance: Curve25519-based cryptography.
+
+## Store path signature format
+
+Store path signatures are ED25519 signatures, there are two types of signatures:
+
+- `ValidPathInfo` will assemble a fingerprint in the form of
+  ```
+  1;<store path>;<NAR hash in base32 including its own type>;<NAR size in bytes>;<NAR references separated by ,>
+  ```
+  and sign this information with the ED25519 private key.
+
+- `Realisation` will assemble a fingerprint in the form of a JSON string:
+  ```
+  { id, outPath, dependentRealisations }
+  ```
+  and sign this information with the ED25519 private key.
+
+# Remote signature protocol
+
+The remote signature protocol is a mechanism to offload the signature of your store paths to another machine that can possess the secret key material in a secure way.
+
+In this setup, Nix will contact a remote signing URL that you specified and ask to sign fingerprints over the wire.
+
+The protocol expects a UNIX domain socket to force you to handle proper authentication and authorization.
+`socat` is a great tool to manipulate all sorts of sockets.
```

Comment:
This is missing an example of how any of this looks in action. Right now I wouldn't even know where to start in order to use it.

Reply:
I wonder if I should really give an example because I don't want to give the trivial example that will make people run it insecurely, so I can give non-trivial examples to run it securely but with imaginary TCP services.

Reply:
Yeah I plan on revising the docs a good bit

Reply:
It would be enough to show how to invoke Nix to listen on a socket. I don't see from the docs which interfaces this change even exposes.